### ISSUE TITLE
number 0 or boolean false do not validate

### ISSUE THREAD

**Issue by JADurham909** [eivindfjeldstad/validate#23]
Numbers with value 0 or booleans set to false will not validate.  I suspect its failing a truthy test in property.validate.

passwordBad: {
            type: 'boolean',
            required: false
        }

validate({
        domain: domainWithPassword,
        user: 'justin',
        password: password,
        passwordBad: false
})

or

passwordBad: {
            type: 'number',
            required: false
        }

validate({
        domain: domainWithPassword,
        user: 'justin',
        password: password,
        passwordBad: 0
})

**Comment by CocaCola183** [comment]
@JADurham909 I got the same problem

**Comment by CocaCola183** [comment]
property.js 

line 153
```
var valid = fn[0].call(ctx, value);
```

line 168~172

```
function required () {
  return function (value) {
    return !!value;
  }
}
```

in this way, if I pass 0(number) or false(bollean), it will return !!0 or !!false

**Comment by eivindfjeldstad** [comment]
hm, yeah thats probably not ideal. maybe something like this instead:
```js
function required () {
  return function (value) {
    return value == null;
  }
}
```

**Comment by eivindfjeldstad** [comment]
The problem is that most people pass in objects constructed from query strings or request bodies. I would expect an empty string to be considered invalid. I'm not sure about 0 though.

**Comment by throoze** [comment]
Hi! I still get the same problem. boolean false does not validate. Will you address this issue in the short term?

**Comment by rlingineni** [comment]
+1 Same problem here. `boolean:false` fails

**Comment by mmitrukaa** [comment]
+1 Same problem , need a fix

**Comment by eivindfjeldstad** [comment]
There's a new branch out for version 4. I'm open to changing the way `.required` works. I agree that `false` should not be considered invalid

**Comment by eivindfjeldstad** [comment]
fixed in 4.0.0
